### DISCUSSION TITLE
Viewport Meta Tag in Cassipeia

### DISCUSSION THREAD

**Discussion by ghost** [joomla/joomla-cms#32847] [Q&A]
In Cassiopeia is no Viewort Meta Tag like

`<meta name="<meta name="viewport" content="width=device-width, initial-scale=1.0" />" content="width=device-width, initial-scale=1.0">`

Is there a reason for this?

**Comment by drmenzelit** [comment]
The meta tag is present in Cassiopeia, see 
https://github.com/joomla/joomla-cms/blob/88b318d467b1c2f97855d0699f704180a7663fbd/templates/cassiopeia/index.php#L95
and 
https://github.com/joomla/joomla-cms/blob/88b318d467b1c2f97855d0699f704180a7663fbd/templates/cassiopeia/index.php#L102
![grafik](https://user-images.githubusercontent.com/9153168/112432322-07e87e80-8d41-11eb-9446-75151894f3f8.png)

**Comment by ghost** [comment]
Thanks, but what is the advantage of this way of adding compared to adding directly as in Protostar?
https://github.com/joomla/joomla-cms/blob/fcacefc380ae0e75da0350eafa1f6f82cf3e98b1/templates/protostar/index.php#L137

**Reply by dgrammatiko** [reply]
> but what is the advantage ...?

Using the API, wherever there is an API, should always be the **right way©**. One very simple, real world example here hopefully will answer your question:
If you hardcode the `<meta name="viewport" content="width=device-width, initial-scale=1.0" /> ` as it was in Protostar you **HAVE NO WAY** to either remove this or edit it (without parsing the final HTML)
If you add it using the proper way `$this->setMetaData('viewport', 'width=device-width, initial-scale=1');` any plugin can remove /edit this meta **WITHOUT** having to parse the whole HTML

In sort **ALWAYS USE THE API**
